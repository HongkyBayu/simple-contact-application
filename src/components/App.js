/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 * @flow
 */

import React, { Component } from 'react';
import Contact from './Contact';
import ContactForm from './ContactForm';
import CONTACT from './DummyContact';
import {
  FlatList,
  StyleSheet,
  View,
  Text,
} from 'react-native';
import SearchContact from './SearchContact';

export default class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      contactData: [],
      searchInput: '',
    };

    this._searchInput = this._searchInput.bind(this);
  }

  componentDidMount() {
    this.setState({ contactData: CONTACT });
  }

  _insertData(data) {
    const { contactData } = this.state;
    this.setState({
      contactData: [ ...contactData, data ],
    });
  }

  _searchInput(value) {
    this.setState({
      searchInput: value,
    });
  }

  _filterList() {
    const contact = this.state.contactData;
    const searchValue = this.state.searchInput;
    const filterList = contact.filter((value) => {
      return value.name.toLowerCase().includes(searchValue.toLowerCase());
    });
    return filterList;
  }

  render() {
    return (
        <View style={styles.container}>
          <Text style={styles.welcome}>
            Contact List
          </Text>
          <SearchContact searchNameInput={this._searchInput}/>
          <FlatList
              style={{ height: 500, paddingTop: 20, }}
              data={this._filterList()}
              keyExtractor={item => item.email}
              renderItem={({ item: { name, email, image } }) => (
                  <Contact
                      contactName={name}
                      contactEmail={email}
                      imageURL={image}
                  />
              )}
          />
          <ContactForm contactFormCallback={(data) => this._insertData(data)}/>
        </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#F5FCFF',
    padding: 20,
  },
  welcome: {
    fontSize: 20,
    textAlign: 'center',
    margin: 10,
  },
  instructions: {
    textAlign: 'center',
    color: '#333333',
    marginBottom: 5,
  },
});
