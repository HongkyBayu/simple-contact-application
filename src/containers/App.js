import React, { Component } from 'react';
import { StackNavigator } from 'react-navigation';
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import LoginPage from '../login/LoginPage';
import ContactList from './ContactList.container';
import rootReducers from '../reducers/index'

const RootStack = StackNavigator(
  {
    LoginPage: {
      screen: LoginPage,
    },
    ContactList: {
      screen: ContactList,
    },
  },
  {
    initialRouteName: 'LoginPage',
  },
);

const store = createStore(rootReducers);

export default class App extends Component {
  render() {
    return (
      <Provider store={store}>
        <RootStack />
      </Provider>
    );
  }
}