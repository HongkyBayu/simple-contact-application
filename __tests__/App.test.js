import 'react-native';
import React from 'react';
import App from '../src/components/App';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

describe('App', () => {
  describe('Render Snapshot', () => {
    it('should render the matching snapshot correctly', () => {
      const app = renderer.create(<App/>).toJSON();
      expect(app).toMatchSnapshot();
    });
  });

  describe('#showData', () => {
    it('should show data based on input state', () => {
      const wrapper = shallow(<App/>);
      wrapper.setState({
        contactData: [
          {
            name: 'Rangga',
            email: 'rangga@gmail.com',
            image: 'rangga.png',
          },
        ],
      });
      const expectedOutput = [
        {
          name: 'Rangga',
          email: 'rangga@gmail.com',
          image: 'rangga.png',
        } ];
      expect(wrapper.state().contactData).toEqual(expectedOutput);
    });

    it('should show data based on input state', () => {
      const wrapper = shallow(<App/>);
      wrapper.setState({
        contactData: [
          {
            name: 'Rangga',
            email: 'rangga@gmail.com',
            image: 'rangga.png',
          },
          {
            name: 'Bayu',
            email: 'Bayu@gmail.com',
            image: 'bayu.png',
          },
        ],
      });
      const expectedOutput = [
        {
          name: 'Rangga',
          email: 'rangga@gmail.com',
          image: 'rangga.png',
        },
        {
          name: 'Bayu',
          email: 'Bayu@gmail.com',
          image: 'bayu.png',
        }, ];
      expect(wrapper.state().contactData).toEqual(expectedOutput);
    });
  });

  describe('#filterList', () => {
    it('should filter the list based on search input from user "Rang"', () => {
      const wrapper = shallow(<App/>);
      wrapper.setState({
        searchInput: 'Rang',
        contactData: [
          {
            name: 'Rangga',
            email: 'rangga@gmail.com',
            image: 'rangga.png',
          },
          {
            name: 'Bayu',
            email: 'Bayu@gmail.com',
            image: 'bayu.png',
          },
        ],
      });
      const expectedOutput = [
        {
          name: 'Rangga',
          email: 'rangga@gmail.com',
          image: 'rangga.png',
        } ];
      const childElemet = wrapper.find('FlatList');
      const childProperty = childElemet.props().data;
      expect(childProperty).toEqual(expectedOutput);
    });

    it('should filter the list based on search input from user "Bay"', () => {
      const wrapper = shallow(<App/>);
      wrapper.setState({
        searchInput: 'Bay',
        contactData: [
          {
            name: 'Rangga',
            email: 'rangga@gmail.com',
            image: 'rangga.png',
          },
          {
            name: 'Bayu',
            email: 'Bayu@gmail.com',
            image: 'bayu.png',
          },
        ],
      });
      const expectedOutput = [
        {
          name: 'Bayu',
          email: 'Bayu@gmail.com',
          image: 'bayu.png',
        } ];
      const childElemet = wrapper.find('FlatList');
      const childProperty = childElemet.props().data;
      expect(childProperty).toEqual(expectedOutput);
    });
  });
});


