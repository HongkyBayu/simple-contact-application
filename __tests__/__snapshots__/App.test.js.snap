// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Render Snapshot should render the matching snapshot correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#F5FCFF",
      "flex": 1,
      "padding": 20,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 20,
        "margin": 10,
        "textAlign": "center",
      }
    }
  >
    Contact List
  </Text>
  <View
    style-={
      Object {
        "alignItems": "flex-start",
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "paddingTop": 10,
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        placeholder="Search"
        style={
          Object {
            "backgroundColor": "#D4D8D8",
            "borderRadius": 10,
            "opacity": 100,
            "textAlign": "center",
          }
        }
      />
    </View>
  </View>
  <RCTScrollView
    data={
      Array [
        Object {
          "email": "hsdpal@gmail.com",
          "image": "https://upload.wikimedia.org/wikipedia/commons/7/7c/Profile_avatar_placeholder_large.png",
          "name": "DP",
        },
        Object {
          "email": "vascodagama1@gmail.com",
          "image": "https://upload.wikimedia.org/wikipedia/commons/7/7c/Profile_avatar_placeholder_large.png",
          "name": "Neel",
        },
        Object {
          "email": "mail@chetankhotari.m",
          "image": "https://upload.wikimedia.org/wikipedia/commons/7/7c/Profile_avatar_placeholder_large.png",
          "name": "Chetan",
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    invertStickyHeaders={undefined}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    style={
      Object {
        "height": 500,
        "paddingTop": 20,
      }
    }
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "alignItems": "flex-end",
              "borderWidth": 0.5,
              "flex": 1,
              "flexDirection": "row",
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "https://upload.wikimedia.org/wikipedia/commons/7/7c/Profile_avatar_placeholder_large.png",
              }
            }
            style={
              Object {
                "height": 60,
                "width": 60,
              }
            }
          />
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 18,
                "height": 44,
                "padding": 10,
              }
            }
          >
            DP
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 10,
                "height": 30,
                "padding": 5,
              }
            }
          >
            hsdpal@gmail.com
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "alignItems": "flex-end",
              "borderWidth": 0.5,
              "flex": 1,
              "flexDirection": "row",
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "https://upload.wikimedia.org/wikipedia/commons/7/7c/Profile_avatar_placeholder_large.png",
              }
            }
            style={
              Object {
                "height": 60,
                "width": 60,
              }
            }
          />
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 18,
                "height": 44,
                "padding": 10,
              }
            }
          >
            Neel
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 10,
                "height": 30,
                "padding": 5,
              }
            }
          >
            vascodagama1@gmail.com
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "alignItems": "flex-end",
              "borderWidth": 0.5,
              "flex": 1,
              "flexDirection": "row",
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "https://upload.wikimedia.org/wikipedia/commons/7/7c/Profile_avatar_placeholder_large.png",
              }
            }
            style={
              Object {
                "height": 60,
                "width": 60,
              }
            }
          />
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 18,
                "height": 44,
                "padding": 10,
              }
            }
          >
            Chetan
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 10,
                "height": 30,
                "padding": 5,
              }
            }
          >
            mail@chetankhotari.m
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "height": 0,
          }
        }
      />
    </View>
  </RCTScrollView>
  <View
    style={
      Object {
        "alignItems": "flex-end",
        "flex": 1,
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "flex": 1,
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        placeholder="Name"
      />
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        placeholder="Email"
      />
    </View>
    <View
      accessibilityComponentType="button"
      accessibilityLabel={undefined}
      accessibilityTraits={
        Array [
          "button",
        ]
      }
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={undefined}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Submit Data
        </Text>
      </View>
    </View>
  </View>
</View>
`;
